static/js/containers/DashboardPage.js

```diff
@@ -371,6 +397,22 @@ class DashboardPage extends React.Component {
     />;
   }
 
+  calculatedPrices = () => {
+    const {
+      dashboard,
+      prices,
+      coupons,
+    } = this.props;
+
+    return calculatePrices(dashboard.programs, prices.coursePrices, coupons.coupons);
+  };
+
+  allPricesZero = () => {
+    let allPricesZero = true;
+    this.calculatedPrices().forEach(price => allPricesZero = allPricesZero && price === 0);
```

Comment:
I think you need to filter on program here, to make sure you're only considering the currently selected program